typing_extensions/src_py3/typing_extensions.py

```diff
@@ -985,6 +997,188 @@ def __new__(cls, *args, **kwds):
         Protocol.__doc__ = Protocol.__doc__.format(bases="Protocol, Generic[T]" if
                                                    OLD_GENERICS else "Protocol[T]")
 
+
+if PEP_560:
+    # We need to rewrite these from scratch.
```

Comment:
Can you elaborate in the comment why? Also, it seems weird that there are now two conditional definitions of `_ProtocolMeta` and `Protocol`, one pair under `if HAVE_PROTOCOLS` above and then this pair. Why not at least execute precisely one pair of definitions at runtime?

Reply:
Yes, this totally makes sense. There is no need to execute both versions in 3.7. I will re-organise the `if`s.